### ISSUE TITLE
TypeError: boardModule is not a constructor

### ISSUE THREAD

**Issue by pleemans67** [monteslu/node-red-contrib-gpio#37]
Hi, I installed the raspi-io plugin in conjunction with the node-red-contrib-gpio.

I get the following error when i try to use one of the nodes:

error initializing io class raspi-io TypeError: boardModule is not a constructor

**Comment by pleemans67** [comment]
maybe to add as information: board = Raspberry Pi Model 3B+

**Comment by Mestryx-dev** [comment]
I have the same problem did you have solve it ? 
Where i can specify "board = Raspberry Pi Model 3B+" ?

Thanks

**Comment by monteslu** [comment]
not sure it'll fix this (haven't tried on a recent pi), but did you install raspi-io as a normal user and run it as root?
https://github.com/nebrius/raspi-io#system-requirements

**Comment by Mestryx-dev** [comment]
when you say run is for node-red run as root ?

**Comment by monteslu** [comment]
yeah unfortunately.  GPIOs from raspi-io can only be accessed with root, so node-red has to be run as root.  The other pi nodes that hit the GPIOs depend on another service that is also running as root.

**Comment by Mestryx-dev** [comment]
Start node-red as Pi user : TypeError: boardModule is not a constructor
in root :  Error: Cannot find module 'raspi-io'

Not work for me.

**Comment by Laca81** [comment]
Hi All!

The problem is the node-red-contrib-gpio is not compatible with the raspi-io 10.0.x. If you downgrade to 9.0.1 or change the nodebotNode.js -> line 249 from:
`node.io = new boardModule();`
to
`node.io = new boardModule.RaspiIO();`

then it should work.

Bests,
Laca

**Comment by monteslu** [comment]
how about something like 

`node.io = boardModule.RaspiIO  ? new boardModule.RaspiIO() : new boardModule()`

or maybe something smarter above that can check for other IO classes that follow the raspio pattern?

Anyone care to make a PR for this?

**Comment by spade69** [comment]
meet the same problem here, my board is PI 4B
